src/controllers/csi/provisioner/agent.go

```diff
@@ -100,9 +100,12 @@ func getUrlProperties(version string) *url.Properties {
 }
 
 func setupImageInstaller(ctx context.Context, fs afero.Fs, pathResolver metadata.PathResolver, apiReader client.Reader, certPath, digest string, dynakube *dynatracev1beta1.DynaKube) (installer.Installer, error) {
-	dockerConfig, err := dockerconfig.NewDockerConfig(ctx, apiReader, *dynakube)
-	if err != nil {
-		return nil, err
+	dockerConfig := dockerconfig.NewDockerConfig(apiReader, *dynakube)
+	if dynakube.Spec.CustomPullSecret != "" {
```

Comment:
maybe refactor in a seperate function like `hasCustomPullSecret`

Reply:
I don't think its necessary, its a higher level property

`dynakube.Spec.TrustedCAs != ""` should also need to be replaced to remain consistent